applications/ParticleMechanicsApplication/custom_strategies/schemes/mpm_explicit_scheme.hpp

```diff
@@ -0,0 +1,565 @@
+                            {
+                                for (IndexType j = 0; j < DomainSize; j++)
+                                {
+                                    r_current_velocity[j] = r_nodal_momenta[j] / nodal_mass;
+                                } // for DomainSize
+                            }
+                        }
+                    }
+                }
+                else if (rCurrentProcessInfo.GetValue(EXPLICIT_STRESS_UPDATE_OPTION) == 2)
+                {
+                    // MUSL stress update. This works by projecting the updated particle
+                    // velocity back to the nodes. The nodal velocity field is then
+                    // used for stress computations.
+
+                    // Reset grid velocities
+                    #pragma omp parallel for
+                    for (int iter = 0; iter < static_cast<int>(rModelPart.Nodes().size()); ++iter)
+                    {
+                        auto i = rModelPart.NodesBegin() + iter;
+                        array_1d<double, 3 >& nodal_velocity = (i)->FastGetSolutionStepValue(VELOCITY);
+                        nodal_velocity.clear();
+                    }
+
+                    // Map updated MP velocities back to grid
+                    const auto it_elem_begin = rModelPart.ElementsBegin();
+                    #pragma omp parallel for
+                    for (int i = 0; i < static_cast<int>(rModelPart.Elements().size()); ++i) {
+                        auto it_elem = it_elem_begin + i;
+                        std::vector<bool> dummy;
+                        it_elem->CalculateOnIntegrationPoints(CALCULATE_MUSL_VELOCITY_FIELD, dummy, rCurrentProcessInfo);
+                    }
+
+                    // Reapply dirichlet BCs to MUSL velocity field
+                    calculateGridVelocityAndApplyDirichletBC(rCurrentProcessInfo);
+                }
+            }
+
+            //***************************************************************************
+            //***************************************************************************
+
+            /** Function that returns the list of Degrees of freedom to be
+            assembled in the system for a Given Element
+             */
+            void GetElementalDofList(
+                Element::Pointer rCurrentElement,
+                Element::DofsVectorType& ElementalDofList,
+                ProcessInfo& CurrentProcessInfo) override
+            {
+                rCurrentElement->GetDofList(ElementalDofList, CurrentProcessInfo);
+            }
+
+            //***************************************************************************
+            //***************************************************************************
+
+            /** Function that returns the list of Degrees of freedom to be
+            assembled in the system for a Given Element
+             */
+            void GetConditionDofList(
```

Comment:
can you please use the new interface?
https://github.com/KratosMultiphysics/Kratos/pull/5290/files#diff-6fd626f1fab184a069188ac76f0fee03R1024-R1028